src/orders/components/OrderRefundReturnAmount/RefundAmountInput.tsx

```diff
@@ -69,11 +69,13 @@ const RefundAmountInput: React.FC<RefundAmountInputProps> = props => {
   const intl = useIntl();
   const classes = useStyles(props);
   const maxRefund = payment.availableRefundAmount;
-  const currentPayment = data.paymentsToRefund?.find(getById(payment.id));
+  const currentPaymentValue = getCurrentPaymentValue(
+    data.paymentsToRefund,
+    payment.id
+  );
   const amountTooBig =
-    Number(currentPayment?.value) > payment.availableRefundAmount?.amount;
-  const amountTooSmall =
-    currentPayment?.value && Number(currentPayment.value) <= 0;
+    Number(currentPaymentValue) > payment.availableRefundAmount?.amount;
```

Comment:
No need to case `currentPaymentValue` to `Number` each time. Just cast it once in the `getCurrentPaymentValue` function. `PriceField` prop `value` can receive both `string` and `number`.

Reply:
`getCurrentPaymentValue` should also return an empty string, not 0.
In this case, it will be impossible to clear the `PriceField` value. It will always be set to 0, even if you want to manually replace 0 with a different amount.